basic/bittwister_test.go

```diff
@@ -0,0 +1,609 @@
+		},
+		{
+			name:             "50Maxms",
+			targetLatency:    50 * time.Millisecond,
+			tolerancePercent: 10,
+		},
+		{
+			name:             "100Maxms",
+			targetLatency:    100 * time.Millisecond,
+			tolerancePercent: 10,
+		},
+		{
+			name:             "200Maxms",
+			targetLatency:    200 * time.Millisecond,
+			tolerancePercent: 10,
+		},
+		{
+			name:             "500Maxms",
+			targetLatency:    500 * time.Millisecond,
+			tolerancePercent: 10,
+		},
+		{
+			name:             "Max1s",
+			targetLatency:    1 * time.Second,
+			tolerancePercent: 10,
+		},
+		{
+			name:             "Max2s",
+			targetLatency:    2 * time.Second,
+			tolerancePercent: 10,
+		},
+		{
+			name:             "Max3s",
+			targetLatency:    3 * time.Second,
+			tolerancePercent: 10,
+		},
+		{
+			name:             "Max5s",
+			targetLatency:    5 * time.Second,
+			tolerancePercent: 10,
+		},
+	}
+
+	targetIP, err := target.GetIP()
+	require.NoError(t, err, "Error getting IP")
+
+	for _, tc := range tt {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			err = target.SetLatencyAndJitter(tc.targetLatency.Milliseconds(), 0)
+			require.NoError(t, err, "Error setting packetloss rate")
+
+			t.Log("Starting latency test. It takes a while.")
+			startTime := time.Now()
+			ctx, cancel := context.WithTimeout(context.Background(), commandTimeout)
+			defer cancel()
+
+			targetAddress := fmt.Sprintf("%s:%d", targetIP, gopingPort)
+			output, err := executor.ExecuteCommandWithContext(ctx,
+				"goping", "ping", "-q",
+				"-c", fmt.Sprint(numOfPingPackets),
+				// we need to make sure the client waits long enough for the server to respond
+				"-t", (packetTimeout + tc.targetLatency).String(),
+				"-m", "latency",
+				targetAddress)
+			require.NoError(t, err, "Error executing command")
+
+			elapsed := time.Since(startTime)
+			t.Logf("Test took %d seconds", int64(elapsed.Seconds()))
+
+			gotLatency, err := time.ParseDuration(output)
+			require.NoError(t, err, "Error parsing output")
+
+			deviationPercent := math.Abs(float64(gotLatency-tc.targetLatency)/float64(tc.targetLatency)) * 100
+			assert.LessOrEqual(t, deviationPercent, float64(tc.tolerancePercent), "Deviation is too high")
+
+			t.Logf("Latency expected: %v \tgot: %v \tdeviation: %.2f%% \ttolerance: %v%%",
+				tc.targetLatency.String(),
+				gotLatency.String(),
+				deviationPercent,
+				tc.tolerancePercent)
+		})
+	}
+}
+func TestBittwister_Jitter(t *testing.T) {
+	t.Parallel()
+	// Setup
+
+	const (
+		numOfPingPackets = 100
+		packetTimeout    = 100 * time.Millisecond
+		commandTimeout   = 10 * time.Minute
+	)
+
+	mother, err := knuu.NewInstance("mother")
+	require.NoError(t, err, "Error creating instance")
+
+	require.NoError(t, mother.EnableBitTwister(), "Error enabling BitTwister")
+	err = mother.SetImage(gopingImage)
+	require.NoError(t, err, "Error setting image")
+
+	gopingPort := 8001
+	err = mother.SetEnvironmentVariable("SERVE_ADDR", fmt.Sprintf("0.0.0.0:%d", gopingPort))
```

Comment:
optimization: If you set the env variable after you commit, the Instance is prepared faster, as it does not need to build a container image.

Reply:
done